src/main/java/io/kestra/plugin/shopify/products/UpdateProduct.java

```diff
@@ -0,0 +1,263 @@
+package io.kestra.plugin.shopify.products;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.io.IOException;
+import java.lang.InterruptedException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.shopify.AbstractShopifyTask;
+import io.kestra.plugin.shopify.models.Product;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.net.URI;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Update an existing product in Shopify store",
+    description = "Update an existing product with new details."
+)
+@Plugin(
+    examples = {
+        @Example(
+        title = "Update product title and price",
+        full = true,
+        code = """
+        id: shopify_update_product
+        namespace: company.team
+
+        tasks:
+          - id: update_product
+            type: io.kestra.plugin.shopify.products.UpdateProduct
+            storeDomain: my-store.myshopify.com
+            accessToken: "{{ secret('SHOPIFY_ACCESS_TOKEN') }}"
+            productId: 123456789
+            title: "Updated T-Shirt Title"
+            bodyHtml: "<p>Updated description for the t-shirt</p>"
+            status: "active"
+        """
+        )
+    }
+)
+public class UpdateProduct extends AbstractShopifyTask implements RunnableTask<UpdateProduct.Output> {
+
+    @Schema(
+        title = "Product ID",
+        description = "The ID of the product to update"
+    )
+    private Property<Long> productId;
```

Comment:
Should be `@NotNull`, check also all the tasks have proper `@NotNull` for mandatory params such as this one (product id for instance)